notifications/src/main/kotlin/uk/gov/govuk/notifications/model/AdditionalData.kt

```
@@ -2,10 +2,23 @@ package uk.gov.govuk.notifications.model
 
 import com.google.gson.Gson
 import com.google.gson.annotations.SerializedName
+import org.json.JSONObject
 
 internal data class AdditionalData(
     @SerializedName("deeplink") val deepLink: String
 )
 
-internal fun String.asAdditionalData(gson: Gson = Gson()) =
-    gson.fromJson(this, AdditionalData::class.java)
+/**
+ * Tries to parse a JSON object to an Additional Data object.
+ *
+ * If successful returns an Additional Data object.
+ * If unsuccessful returns null.
+ */
+internal fun JSONObject?.asAdditionalData(): AdditionalData? {
+    return try {
+        val gson = Gson()
+        gson.fromJson(this.toString(), AdditionalData::class.java)
+    } catch (_: Exception) {
+        null
```

Do we need any form of indication if this is happening a lot in order to flag issues with the creation of the notifications?

---

Both iOS and Android fail silently if they cannot parse the additional data. You've raised a good point which I am going to raise separate iOS and Android tickets for and take to design so they can decide the behaviour if they want Page Not Found or something else.